descriptor.md

```diff
@@ -0,0 +1,113 @@
+<!--[metadata]>
++++
+draft = true
++++
+<![end-metadata]-->
+
+# OpenContainers Content Descriptors
+
+OCI have several components that come to together to describe an image. 
+References between components form a [Merkle Directed Acyclic Graph (DAG)](https://en.wikipedia.org/wiki/Merkle_tree).
+The references in the _Merkle DAG_ are expressed through _Content Descriptors_.
+A _Content Descriptor_ or _Descriptor_, describes the disposition of targeted content.
+A _Descriptor_ includes the type of content, an independently-verifiable content identifier, known as a "digest" and the byte-size of the raw content.
+
+Descriptors SHOULD be embedded in other formats to securely reference external content.
+
+Descriptors MAY be used as an independent object to specify content.
+
+Implementations MUST verify fetched content against 
+
+## Properties
+
+- **`mediaType`** *string*
+
+  This REQUIRED property contains the MIME type of the referenced object.
+
+- **`digest`** *string*
+
+  This REQUIRED parameter is the digest of the content, meeting the requirements outlined in [Digests and Verification](#digests-and—verification).
+  Retrieved content SHOULD be verified against this digest before processing.
+
+- **`size`** *int*
+  This REQUIRED property specifies the size in bytes of the object.
+  This property exists so that a client will have an expected size for the content before validating.
+  If the length of the retrieved content does not match the specified length, the content should not be trusted.
+
+### Reserved
```

Comment:
On Thu, Jun 02, 2016 at 03:33:16PM -0700, Stephen Day wrote:
> Either way, this clause wouldn't protect against vendor-added fields
> in a descriptor. That is simply not supported. If one drops a random
> field into a descriptor, it may be overwritten by future versions of
> the specification.

That *is* supported by JSON-LD.  Here's how you use both at once:

  {
    "@context": [
      "http://opencontainers.org/image/descriptor/1.0",
      {
        "vendorUrls": "http://example.com/overload/urls"
      }
    ],
    "mediaType": "application/vnd.oci.image.manifest.v1+json",
    "size": 7682,
    "digest": "sha256:5b0bcabd1ed22e9fb1310cf6c2dec7cdef19f0ad69efa1f392e94a4333501270",
    "urls": [
      "https://a.example.com/cas",
      "https://b.example.com/blobs"
    ],
    "vendorUrls": ["a.example.com"]
  }

With JSON-LD, the property name (e.g. ‘urls’, ‘vendorUrls’, …) isn't
the semantic identifier; it's just a link to the @context entry where
you assign semantic IRIs
(e.g. ‘http://opencontainers.org/image/descriptor/urls’,
http://example.com/overload/urls’, …)